cpp/src/dictionary/remove_keys.cu

```diff
@@ -99,15 +104,15 @@ std::unique_ptr<column> remove_keys_fn(
   std::unique_ptr<column> indices_column(std::move(table_indices.front()));
 
   // compute new nulls -- merge the existing nulls with the newly created ones (value<0)
-  auto d_null_mask = dictionary_column.null_mask();
-  auto d_indices   = indices_column->view().data<int32_t>();
-  auto new_nulls   = cudf::detail::valid_if(
-    thrust::make_counting_iterator<size_type>(dictionary_column.offset()),
-    thrust::make_counting_iterator<size_type>(dictionary_column.offset() +
-                                              dictionary_column.size()),
-    [d_null_mask, d_indices] __device__(size_type idx) {
-      if (d_null_mask && !bit_is_set(d_null_mask, idx)) return false;
-      return (d_indices[idx] >= 0);  // new nulls have negative values
+  auto const offset = dictionary_column.offset();
+  auto d_null_mask  = dictionary_column.null_mask();
+  auto d_indices    = indices_column->view().data<uint32_t>();
+  auto new_nulls    = cudf::detail::valid_if(
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(dictionary_column.size()),
+    [offset, d_null_mask, d_indices] __device__(size_type idx) {
+      if (d_null_mask && !bit_is_set(d_null_mask, idx + offset)) return false;
+      return (d_indices[idx] < std::numeric_limits<size_type>::max());  // new nulls have max values
```

Comment:
similarly 
```suggestion
      return (d_indices[idx] < std::numeric_limits<uint32_t>::max());  // new nulls have max values
```

Reply:
Same answer. Regardless of the type, the max number of elements is still max of `size_type`.